### PR TITLE
generate pki with terraform

### PR DESCRIPTION
**Pull request by mikedanese**
I think this is an improvement. It reduces moving parts and moves kubeconfig generation out of phase2. This also improves the security of the GCE setup by requiring ComputeReadScope to see the apiserver private key (instead of StorateReadScope) and completely removing the ability to see the root CA private key (instead of requiring StorageReadScope (i know, it's bad)). Please review commits individually.

### CODE REVIEW

**Comment by mikedanese** [comment]
@errordeveloper @colemickens this is working, please review.

File: phase1/gce/configure-vm.sh [hunk 1]
```diff
@@ -4,12 +4,14 @@ set -o errexit
 set -o pipefail
 set -o nounset
 
-ROLE=$(curl \
-  -H "Metadata-Flavor: Google" \
-  "metadata/computeMetadata/v1/instance/attributes/k8s-role")
-BUCKET=$(curl \
-  -H "Metadata-Flavor: Google" \
-  "metadata/computeMetadata/v1/instance/attributes/k8s-deploy-bucket")
+get_metadata() {
```
Comment (errordeveloper) [inline phase1/gce/configure-vm.sh]: Just had quick thought... Would fetching and parsing these from Ansible be reasonable in terms of phase separation?

Comment (mikedanese) [inline phase1/gce/configure-vm.sh]: We need to etch this out when we figure out the contract. This requires a switch on which phase 1 you are using (different impl per cloud provider) so it's convienient to roll it into the phase1 handoff. This make phase2 a bit more genenral.

**Comment by mikedanese** [comment]
In the spirit of moving quickly I'm going to merge and build on this.

**Comment by errordeveloper** [comment]
+1

**Pull request merged**